Assets/_Scripts/Actions/ActionBase.cs

```diff
@@ -172,16 +179,20 @@ public class ActionBase : ScriptableObject {
         return true;
     }
 
-    public FightResult ExecuteAction(FightingEntity user, List<FightingEntity> targets) {
+    public IEnumerator ExecuteAction(FightingEntity user, List<FightingEntity> targets) {
         if (!CheckCost(user)) {
-            return new FightResult(user, this);
+            lastFightResult = new FightResult(user, this);
+            yield break;
```

Comment:
I'm not particularly fond of using member variables as "return" values, but if its only being ingested by DoFight, it should be fine.